OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -743,9 +745,49 @@ void UpdatePlayerList()
 			tabCompletion.Names = orderManager.LobbyInfo.Clients.Select(c => c.Name).Distinct().ToList();
 		}
 
+		void UpdateDiscordStatus()
+		{
+			string secret = null;
+			if (orderManager.LobbyInfo.GlobalSettings.Dedicated)
+			{
+				var endpoint = orderManager.Endpoint.GetConnectEndPoints().First();
+				secret = string.Concat(endpoint.Address, "|", endpoint.Port);
+			}
+
+			var state = skirmishMode ? DiscordState.InSkirmishLobby : DiscordState.InMultiplayerLobby;
+			DiscordService.UpdateStatus(state, secret: secret);
```

Comment:
From IRC: we probably don't want to be updating this every time the lobby info changes, only when the `Dedicated` flag changes.

Reply:
Was not a problem, cuz we have a check in DiscordService.UpdateStatus.